engines/gargoyle/detection.cpp

```diff
@@ -0,0 +1,323 @@
+	return (bool)(_gameDescription->_desc.flags & ADGF_DEMO);
+}
+
+Common::Language GargoyleEngine::getLanguage() const {
+	return _gameDescription->_desc.language;
+}
+
+InterpreterType GargoyleEngine::getInterpreterType() const {
+	return _gameDescription->_interpType;
+}
+
+const Common::String &GargoyleEngine::getGameMD5() const {
+	return _gameDescription->_md5;
+}
+
+} // End of namespace Gargoyle
+
+#include "gargoyle/frotz/detection_tables.h"
+#define ZCODE(ID, NAME) { ID, Gargoyle::Frotz::NAME##_DESC }
+
+static const PlainGameDescriptor gargoyleGames[] = {
+	{"zcode", "Zcode Games" },
+	{"scottadams", "Scott Adams Games"},
+
+	// Infocom/Z-code games
+	ZCODE("amfv", AMFV),
+	ZCODE("arthur", ARTHUR),
+	ZCODE("ballyhoo", BALLYHOO),
+	ZCODE("beyondzork", BEYONDZORK),
+	ZCODE("borderzone", BORDERZONE),
+	ZCODE("bureaucracy", BUREAUCRACY),
+	ZCODE("cutthroats", CUTTHROATS),
+	ZCODE("deadline", DEADLINE),
+	ZCODE("enchanter", ENCHANTER),
+	ZCODE("hhgttg", HHGTTG),
+	ZCODE("hijinx", HIJINX),
+	ZCODE("infidel", INFIDEL),
+	ZCODE("journey", JOURNEY),
+	ZCODE("lgop", LGOP),
+	ZCODE("lgop2", LGOP2),
+	ZCODE("lurking", LURKING),
+	ZCODE("minizork1", MINIZORK1),
+	ZCODE("moonmist", MOONMIST),
+	ZCODE("nordbert", NORDBERT),
+	ZCODE("planetfall", PLANETFALL),
+	ZCODE("plundered", PLUNDERED),
+	ZCODE("sampler1", SAMPLER1),
+	ZCODE("sampler2", SAMPLER2),
+	ZCODE("seastalker", SEASTALKER),
+	ZCODE("sherlockriddle", SHERLOCKRIDDLE),
+	ZCODE("shogun", SHOGUN),
+	ZCODE("sorcerer", SORCERER),
+	ZCODE("spellbreaker", SPELLBREAKER),
+	ZCODE("starcross", STARCROSS),
+	ZCODE("stationfall", STATIONFALL),
+	ZCODE("suspect", SUSPECT),
+	ZCODE("suspended", SUSPENDED),
+	ZCODE("trinity", TRINITY),
+	ZCODE("wishbringer", WISHBRINGER),
+	ZCODE("witness", WITNESS),
+	ZCODE("zork0", ZORK0),
+	ZCODE("zork1", ZORK1),
+	ZCODE("zork2", ZORK2),
+	ZCODE("zork3", ZORK3),
+	ZCODE("ztuu", ZTUU),
+
+	// Scott Adams games
+	{ "adventureland", "Adventureland" },
+	{ "pirateadventure", "Pirate Adventure" },
+	{ "missionimpossible", "Mission Impossible" },
+	{ "voodoocastle", "Voodoo Castle" },
+	{ "thecount", "The Count" },
+	{ "strangeodyssey", "Strange Odyssey" },
+	{ "mysteryfunhouse", "Mystery Fun House" },
+	{ "pyramidofdoom", "Pyramid Of Doom" },
+	{ "ghosttown", "Ghost Town" },
+	{ "savageisland1", "Savage Island, Part 1" },
+	{ "savageisland2", "Savage Island, Part 2" },
+	{ "goldenvoyage", "The Golden Voyage" },
+	{ "adventure13", "Adventure 13" },
+	{ "adventure14", "Adventure 14" },
+	{ "buckaroobonzai", "Buckaroo Banzai" },
+	{ "marveladventure", "Marvel Adventure #1" },
+	{ "scottsampler", "Adventure International's Mini-Adventure Sampler" },
+	{0, 0}
+};
+
+#include "common/config-manager.h"
+#include "common/file.h"
+#include "gargoyle/detection_tables.h"
+#include "gargoyle/frotz/detection.h"
+#include "gargoyle/frotz/frotz.h"
+#include "gargoyle/scott/detection.h"
+#include "gargoyle/scott/scott.h"
+
+class GargoyleMetaEngine : public AdvancedMetaEngine {
```

Comment:
IMO, this class should either extend `AdvancedMetaEngine` and use its standard detection code or extend `MetaEngine` with the current detection code.

Reply:
I'll have a look into extending from MetaEngine instead then.